ui/src/components/dataset/ingestion/IngestionStepper.vue

```diff
@@ -682,61 +1010,178 @@ const onSubmit = async () => {
+const fetchAssociatedProject = async () => {
+  // If associated Project has already been fetched, skip.
+  if ((createdDataset.value.projects || []).length > 0) {
+    return;
+  }
+  // Fetch associated Projects.
+  try {
+    const res = await datasetService.getById({
+      id: createdDataset.value.id,
+      include_projects: true,
+    });
+    createdDataset.value.projects = res.data.projects;
+  } catch (error) {
+    // console.error("Error fetching associated projects:", error);
+    throw new Error(ERRORS.FETCH_ASSOCIATED_PROJECTS);
+  }
+};
+
+// This method is expected to be idempotent, to ensure that it can be called upon form-submission retries.
+const fetchAssociatedProjectDetails = async () => {
+  // If associated Project's details has already been fetched, skip.
+  if (projectCreated.value) {
+    return;
+  }
+  // Fetch associated Project's details.
+  try {
+    const res = await projectService.getById({
+      id: createdDataset.value.projects[0].project_id,
+      forSelf: !(auth.canOperate || auth.canAdmin),
+    });
+    projectCreated.value = res.data;
+  } catch (error) {
+    // console.error("Error fetching associated project details:", error);
+    throw new Error(ERRORS.GET_PROJECT_DETAILS);
+  }
+};
+
+const initiateIngestion = async () => {
+  try {
+    await datasetService.initiate_workflow_on_dataset({
+      dataset_id: createdDataset.value.id,
+      workflow: "integrated",
+    });
+    toast.success("Initiated dataset ingestion");
+  } catch (error) {
+    // console.error("Error initiating ingestion:", error);
+    throw new Error(ERRORS.INITIATE_INGESTION);
+  }
+};
+
+const handleSuccessfulIngestion = () => {
+  submissionButtonText.value = SUBMIT_BUTTON.INGEST;
+  submissionSuccess.value = true;
+};
+
+const handleSubmissionError = (error) => {
+  // console.error("Error during submission:", error);
+  let errorMessage;
+
+  switch (error.message) {
+    case ERRORS.CREATE_DATASET:
+    case ERRORS.FETCH_ASSOCIATED_PROJECTS:
+    case ERRORS.GET_PROJECT_DETAILS:
+    case ERRORS.INITIATE_INGESTION:
+      errorMessage = "An error occurred. Please try again.";
+      break;
+    case ERRORS.DATASET_EXISTS:
+      errorMessage = ERRORS.DATASET_EXISTS;
+      break;
+    default:
+      errorMessage =
+        error.message || "An unexpected error occurred. Please try again.";
+  }
+
+  toast.error(errorMessage);
+  submissionButtonText.value = SUBMIT_BUTTON.RETRY;
+};
+
+const ERRORS = {
+  CREATE_DATASET: "Failed to create Dataset",
+  DATASET_EXISTS: "A Dataset with this name already exists",
+  FETCH_ASSOCIATED_PROJECTS: "Failed to fetch associated Project",
+  GET_PROJECT_DETAILS: "Failed to retrieve associated Project's details",
+  INITIATE_INGESTION: "Failed to initiate Ingestion",
 };
 
 const onNextClick = (nextStep) => {
   if (isLastStep.value) {
-    onSubmit();
+    if (submitAttempted.value && !submissionSuccess.value) {
+      if (submissionButtonText.value === SUBMIT_BUTTON.RETRY) {
```

Comment:
Track if form-submission needs to start from beginning or from last failed step. Modified `onSubmit()` definition accordingly.